File: app/src/main/java/org/cuieney/videolife/presenter/contract/MusicHomeContract.java
package org.cuieney.videolife.presenter.contract;

import org.cuieney.videolife.common.base.BasePresenter;
import org.cuieney.videolife.common.base.BaseView;
import org.cuieney.videolife.entity.MusicListBean;
import org.cuieney.videolife.entity.VideoListBean;

import java.util.List;

/**
 * Created by cuieney on 17/3/4.
 */

public interface MusicHomeContract {
    interface View extends BaseView {
        void showContent(List<MusicListBean> musicListBean);
        void error(Throwable throwable);
    }

    interface Presenter extends BasePresenter<MusicHomeContract.View> {
        void getMusicData(String date);

    }
}

File: app/src/main/java/org/cuieney/videolife/kotlin/presenter/contract/MusicHomeContract.kt
package org.cuieney.videolife.kotlin.presenter.contract

import org.cuieney.videolife.entity.MusicListBean
import org.cuieney.videolife.kotlin.base.BasePresenter
import org.cuieney.videolife.kotlin.base.BaseView

/**
 * Created by cuieney on 2017/6/5.
 */
interface MusicHomeContract {
    interface View : BaseView {
        fun showContent(musicListBean: MutableList<MusicListBean>)
        fun error(throwable: Throwable)
    }

    interface Presenter : BasePresenter {
        fun getMusicData(date: String, view: View)
    }
}